src/main/java/gregtech/api/capability/IAdvancedFluidContainer.java

```diff
@@ -0,0 +1,117 @@
+package gregtech.api.capability;
+
+import gregtech.api.fluids.fluid.IExtendedFluid;
+import gregtech.api.fluids.info.FluidState;
+import gregtech.api.fluids.info.FluidTag;
+import gregtech.api.fluids.info.FluidTags;
+import gregtech.api.util.GTUtility;
+import gregtech.client.utils.TooltipHelper;
+import net.minecraft.client.resources.I18n;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Interface defining the abilities of a GT fluid container
+ */
+@FunctionalInterface
+public interface IAdvancedFluidContainer {
+
+    default boolean canHoldFluid(@Nullable FluidStack stack) {
+        if (stack == null || stack.getFluid() == null) return false;
+
+        Fluid fluid = stack.getFluid();
+        if (fluid instanceof IExtendedFluid) {
+            IExtendedFluid advanced = (IExtendedFluid) fluid;
+            FluidState state = advanced.getState();
+            if (!this.canHandleState(state)) return false;
+            if (!this.canHandleTemperature(state, fluid.getTemperature(stack))) return false;
+
+            for (FluidTag tag : advanced.getTags()) {
+                if (tag.requiresContainmentCheck() && !this.canHandleTag(tag)) return false;
+            }
+        } else {
+            FluidState state = FluidState.LIQUID;
+            if (fluid.getUnlocalizedName().contains("plasma")) state = FluidState.PLASMA;
```

Comment:
Is this to catch plasmas from other mods? As currently all our fluids are instance of IExtendedFluid, so none of the plasmas from our mods would be caught by this.